### PR TITLE
Feature/symfony form type range support

### PR DESCRIPTION
**Pull request by vansari**
Changed the argument name of the methods 'resolveFormType' and 'resolveFormOptions' from schema to property for more clearness.

The Unit tests were expanded about a Exception test and new the implemented 'RangeType'.

### CODE REVIEW

File: src/Form/Helper/FormHelper.php [hunk 1]
```diff
@@ -11,39 +12,43 @@
...
 
 class FormHelper
 {
     /**
-     * @param object $schema
+     * @param stdClass $property
      * @return string|null
      */
-    public static function resolveFormType($schema): ?string
+    public static function resolveFormType(stdClass $property): ?string
```
Comment (cyve) [inline src/Form/Helper/FormHelper.php]:
```suggestion
    public static function resolveFormType(\stdClass $property): ?string
```
and remove `use stdClass;`

File: src/Form/Helper/FormHelper.php [hunk 2]
```diff
@@ -11,39 +12,43 @@
...
+        switch ($property->type) {
             case 'array':
                 return CollectionType::class;
             case 'object':
                 return SchemaType::class;
             case 'integer':
                 return IntegerType::class;
             case 'number':
+                if (static::isRangeType($property)) {
+                    return RangeType::class;
```
Comment (cyve) [inline src/Form/Helper/FormHelper.php:44]: 👍

File: src/Form/Helper/FormHelper.php [hunk 3]
```diff
@@ -116,8 +121,37 @@ public static function resolveFormOptions($schema): array
                     default:
                         return $options;
                 }
+            case 'number':
+                switch (true) {
+                    case static::isRangeType($property):
+                        return $options + [
+                                'attr' => [
+                                    'min' => $property->minimum ?? $property->exclusiveMinimum,
```
Comment (cyve) [inline src/Form/Helper/FormHelper.php:128]: Shouldn't it be `$property->minimum ?? ($property->exclusiveMinimum +1)`

File: tests/Form/Helper/FormHelperTest.php [hunk 4]
```diff
@@ -13,18 +15,35 @@
...
+     * @covers ::resolveFormType
      * @dataProvider resolveFormTypeDataProvider
+     * @param mixed $property
+     * @param mixed $expected
      */
-    public function testResolveFormType($schema, $expected)
+    public function testResolveFormType($property, $expected)
     {
-        $this->assertEquals($expected, FormHelper::resolveFormType($schema));
+        if (is_a($expected, LogicException::class, true)) {
```
Comment (cyve) [inline tests/Form/Helper/FormHelperTest.php]:
```suggestion
        if (is_a($expected, \LogicException::class, true)) {
```
And remove `use LogicException;` (and other native classes)
Testing for exceptions is a very good point, thx 👍

**Comment by cyve** [comment]
Hi @vansari, thank you very much for your contribution :-)
I made a few comments

File: src/Form/Helper/FormHelper.php (hunk 3 above)
Comment (vansari) [reply src/Form/Helper/FormHelper.php:128]: The problem is that the attribute exclusiveMinimum / -Maximum has another logic in draft-04 as in draft-07. In draft-04 the attribute accept a boolean value while in draft-07 it accepts a number value. Sure it is possible to create another checkMethod to identify whether the value of the attribute is a boolean or a number, but I think this should be make in another PR. 😄

Comment (cyve) [reply src/Form/Helper/FormHelper.php:128]: Ok

**Pull request merged**